test/powershell/Modules/Microsoft.PowerShell.Core/Enter-PSHostProcess.Tests.ps1

```diff
@@ -2,54 +2,106 @@
 # Licensed under the MIT License.
 
 Describe "Enter-PSHostProcess tests" -Tag Feature {
-    BeforeAll {
-        $pwsh_started = New-TemporaryFile
-        $si = [System.Diagnostics.ProcessStartInfo]::new()
-        $si.FileName = "pwsh"
-        $si.Arguments = "-noexit -command 'pwsh' > '$pwsh_started'"
-        $si.RedirectStandardInput = $true
-        $si.RedirectStandardOutput = $true
-        $si.RedirectStandardError = $true
-        $pwsh = [System.Diagnostics.Process]::Start($si)
-
-        if ($IsWindows) {
-            $powershell_started = New-TemporaryFile
-            $si.FileName = "powershell"
-            $si.Arguments = "-noexit -command 'powershell' >'$powershell_started'"
-            $powershell = [System.Diagnostics.Process]::Start($si)
+    Context "By Process Id" {
+        BeforeAll {
+            $pwsh_started = New-TemporaryFile
+            $si = [System.Diagnostics.ProcessStartInfo]::new()
+            $si.FileName = "pwsh"
+            $si.Arguments = "-noexit -command 'pwsh' > '$pwsh_started'"
+            $si.RedirectStandardInput = $true
+            $si.RedirectStandardOutput = $true
+            $si.RedirectStandardError = $true
+            $pwsh = [System.Diagnostics.Process]::Start($si)
+
+            if ($IsWindows) {
+                $powershell_started = New-TemporaryFile
+                $si.FileName = "powershell"
+                $si.Arguments = "-noexit -command 'powershell' >'$powershell_started'"
+                $powershell = [System.Diagnostics.Process]::Start($si)
+            }
+
         }
 
-    }
+        AfterAll {
+            $pwsh | Stop-Process
+            Remove-Item $pwsh_started -Force -ErrorAction SilentlyContinue
 
-    AfterAll {
-        $pwsh | Stop-Process
-        Remove-Item $pwsh_started -Force -ErrorAction SilentlyContinue
+            if ($IsWindows) {
+                $powershell | Stop-Process
+                Remove-Item $powershell_started -Force -ErrorAction SilentlyContinue
+            }
+        }
+
+        It "Can enter and exit another PSHost" {
+            Wait-UntilTrue { (Get-PSHostProcessInfo -Id $pwsh.Id) -ne $null }
 
-        if ($IsWindows) {
-            $powershell | Stop-Process
-            Remove-Item $powershell_started -Force -ErrorAction SilentlyContinue
+            "enter-pshostprocess -id $($pwsh.Id) -erroraction stop
+            `$pid
+            exit-pshostprocess" | pwsh -c - | Should -Be $pwsh.Id
```

Comment:
Could you please use right case for commends in the string.  `Enter-PSHostProcess -Id $($pwsh.Id) -ErrorAction Stop`

Below too.

Reply:
I'm going to blame this one on @SteveL-MSFT 😜 

Will fix.